mlpp_lib/model_selection.py

```diff
@@ -1,10 +1,329 @@
+            }
+        else:
+            indexers = self.partitions[partition]
+
+        res = tuple(ds.sel(indexers) for ds in args)
+        for ds in args:
+            ds.close()
+        del args
+        return res
+
+    def _time_partitioning(self) -> None:
+
+        # actual computation of splits
+
+        if self._time_defined: # provided time split is all valid xarray indexers (lists or slices)
+            self._time_indexers = self.time_split
+        else:
+            self._time_indexers = {}
+            if all([isinstance(v, float) for v in self.time_split.values()]): # only fractions
+                res = self._time_partition_method(self.time_split)
+                self._time_indexers.update(res)
+            else: # mixed
+                _time_split = self.time_split.copy()
+                self._time_indexers.update({"test": _time_split.pop("test")})
+                res = self._time_partition_method(_time_split)
+                self._time_indexers.update(res)
+
+        # assign indexers
+        for partition in self.partition_names:
+            idx = self._time_indexers[partition]
+            idx = slice(*idx) if len(idx) == 2 else idx 
+            indexer = {"forecast_reference_time": idx}
+            if not hasattr(self, "partitions"):
+                self.partitions = {p: {} for p in self.partition_names}
+            self.partitions[partition].update(indexer)
+
+    def _time_partition_method(self, fractions: Mapping[str, float]):
+        if self.time_split_method == "sequential":
+            return sequential_split(self.time_index, fractions)
+
+    def _station_partitioning(self):
+        """
+        Compute station partitioning for this DataSplitter instance.
+        """
+
+        self._station_indexers: dict[str, xindex] = {}
+        if not self._station_defined:
+            if all([isinstance(v, float) for v in self.station_split.values()]):
+                res = self._station_partition_method(self.station_split)
+                self._station_indexers.update(res)
+            else:
+                _station_split = self.station_split.copy()
+                self._station_indexers.update({"test": _station_split.pop("test")})
+                res = self._station_partition_method(_station_split)
+                self._station_indexers.update(res)
+        else:
+            self._station_indexers.update(self.station_split)
+
+        for partition in self.partition_names:
+            indexer = {"station": self._station_indexers[partition]}
+            if not hasattr(self, "partitions"):
+                self.partitions = {p: {} for p in self.partition_names}
+            self.partitions[partition].update(indexer)
+
+    def _station_partition_method(
+        self, fractions: Mapping[str, float]
+    ) -> Mapping[str, np.ndarray]:
+
+        if self.station_split_method == "random":
+            out = random_split(self.station_index, fractions)
```

Comment:
what about the stratified random split that we use for stations?

Reply:
I introduced this new class but haven't focused on integrating it with existing code yet. We could maybe leave it for another PR, what do you think? For now, one can simply use the existing methods beforehand and then pass the lists/arrays of indexes as `time_split` or `station_split`.